src/mame/nintendo/n64_v.cpp

```diff
@@ -1358,368 +1360,370 @@ void n64_rdp::disassemble(uint64_t *cmd_buf, char* buffer)
 	const uint32_t command = (cmd_buf[0] >> 56) & 0x3f;
 	switch (command)
 	{
-		case 0x00:  sprintf(buffer, "No Op"); break;
+		case 0x00:  buffer << "No Op"; break;
 		case 0x08:      // Tri_NoShade
 		{
 			const int32_t lft = (cmd_buf[0] >> 55) & 0x1;
 
-			sprintf(yl,     "%4.4f", (float)((cmd_buf[0] >> 32) & 0x1fff) / 4.0f);
-			sprintf(ym,     "%4.4f", (float)((cmd_buf[0] >> 16) & 0x1fff) / 4.0f);
-			sprintf(yh,     "%4.4f", (float)((cmd_buf[0] >>  0) & 0x1fff) / 4.0f);
-			sprintf(xl,     "%4.4f", (float)int32_t(cmd_buf[1] >> 32) / 65536.0f);
-			sprintf(dxldy,  "%4.4f", (float)int32_t(cmd_buf[1])       / 65536.0f);
-			sprintf(xh,     "%4.4f", (float)int32_t(cmd_buf[2] >> 32) / 65536.0f);
-			sprintf(dxhdy,  "%4.4f", (float)int32_t(cmd_buf[2])       / 65536.0f);
-			sprintf(xm,     "%4.4f", (float)int32_t(cmd_buf[3] >> 32) / 65536.0f);
-			sprintf(dxmdy,  "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
-
-			sprintf(buffer, "Tri_NoShade            %d, XL: %s, XM: %s, XH: %s, YL: %s, YM: %s, YH: %s\n", lft, xl,xm,xh,yl,ym,yh);
+			snprintf(yl,     sizeof(yl),    "%4.4f", (float)((cmd_buf[0] >> 32) & 0x1fff) / 4.0f);
+			snprintf(ym,     sizeof(ym),    "%4.4f", (float)((cmd_buf[0] >> 16) & 0x1fff) / 4.0f);
+			snprintf(yh,     sizeof(yh),    "%4.4f", (float)((cmd_buf[0] >>  0) & 0x1fff) / 4.0f);
+			snprintf(xl,     sizeof(xl),    "%4.4f", (float)int32_t(cmd_buf[1] >> 32) / 65536.0f);
+			snprintf(dxldy,  sizeof(dxldy), "%4.4f", (float)int32_t(cmd_buf[1])       / 65536.0f);
+			snprintf(xh,     sizeof(xh),    "%4.4f", (float)int32_t(cmd_buf[2] >> 32) / 65536.0f);
+			snprintf(dxhdy,  sizeof(dxhdy), "%4.4f", (float)int32_t(cmd_buf[2])       / 65536.0f);
+			snprintf(xm,     sizeof(xm),    "%4.4f", (float)int32_t(cmd_buf[3] >> 32) / 65536.0f);
+			snprintf(dxmdy,  sizeof(dxmdy), "%4.4f", (float)int32_t(cmd_buf[3])       / 65536.0f);
+
+			util::stream_format(buffer, "Tri_NoShade            %d, XL: %s, XM: %s, XH: %s, YL: %s, YM: %s, YH: %s\n", lft, xl,xm,xh,yl,ym,yh);
```

Comment:
We could make yl, ym etc float types and do the "%4.4f" formatting in the stream_format line?
Or it's possibly cleaner as is?
Or local scope defined `auto yl = util::string_format("%4.4f", ...`

Reply:
I'll address in a new PR